### PR TITLE
FC: Detect banking app installed case

### PR DESCRIPTION
**Pull request by vardges-stripe**
## Summary

Don't open auth urls in safari if app is not installed
## Motivation

Makes for a better UX.

This one is a bit tricky. When we detect the app is not installed, we clear app return url from server then open the non native redirect link.

## Testing

See how I first go with app installed, then delete and try again and it opens in webview
https://user-images.githubusercontent.com/92807969/226878470-735358bd-a5c2-41af-ab93-21e84b320543.MP4

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift [hunk 1]
```diff
@@ -63,6 +64,34 @@ final class PartnerAuthDataSourceImplementation: PartnerAuthDataSource {
         }
     }
 
+    func clearReturnURL(authSession: FinancialConnectionsAuthSession, authURL: String) -> Future<FinancialConnectionsAuthSession> {
+        let promise = Promise<FinancialConnectionsAuthSession>()
+
+        apiClient
+            .generateSessionManifest(clientSecret: clientSecret, returnURL: nil)
+            .observe { [weak self] result in
+                guard let self = self else { return }
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift:73]: This is idiotic a bit, but is reasonable to clear the server state.

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift [hunk 2]
```diff
@@ -63,6 +64,34 @@ final class PartnerAuthDataSourceImplementation: PartnerAuthDataSource {
...
+                switch result {
+                case .success:
+                    let copiedSession = FinancialConnectionsAuthSession(id: authSession.id,
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift:76]: Hacky, I do this only to change the url to not have the prefix

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift [hunk 3]
```diff
@@ -63,6 +64,34 @@ final class PartnerAuthDataSourceImplementation: PartnerAuthDataSource {
...
+                                                                        flow: authSession.flow,
+                                                                        institutionSkipAccountSelection: authSession.institutionSkipAccountSelection,
+                                                                        nextPane: authSession.nextPane,
+                                                                        showPartnerDisclosure: authSession.showPartnerDisclosure,
+                                                                        skipAccountSelection: authSession.skipAccountSelection,
+                                                                        url: authURL,
+                                                                        isOauth: authSession.isOauth,
+                                                                        display: authSession.display)
+                    self.pendingAuthSession = copiedSession
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift:85]: @kgaidis-stripe is this needed? I think so?

Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift:85]: seems reasonable

**Review by kgaidis-stripe — APPROVED** [review]
🎉  great to see this working

**Pull request merged**